smart_open/tests/test_smart_open.py

```diff
@@ -1840,6 +1841,176 @@ def test(self):
+    # ----------- | ---------- | -------- |
+    # 'extension' | False      | Enable   |
+    # 'none'      | False      | Disable  |
+    @parameterizedtestcase.ParameterizedTestCase.parameterize(
+        ("_compression", "decompressor", "compression_kwargs", "no_compression_kwargs"),
+        [
+            (
+                "gz",
+                gzip.decompress,
+                dict(compression=INFER_FROM_EXTENSION),
+                dict(compression=NO_COMPRESSION),
+            ),
+            (
+                "bz2",
+                bz2.decompress,
+                dict(compression=INFER_FROM_EXTENSION),
+                dict(compression=NO_COMPRESSION),
+            )
+        ],
+    )
+    @mock_s3
+    def test_rw_compression_by_extension(
+        self, _compression, decompressor, compression_kwargs, no_compression_kwargs
+    ):
+        """Should read/write files with `_compression`, explicitily inferred by file extension."""
+        s3 = boto3.resource("s3")
+        s3.create_bucket(Bucket="bucket")
+        key = f"s3://bucket/key.{_compression}"
+
+        with smart_open.open(key, "wb", **compression_kwargs) as fout:
+            fout.write(_RAW_DATA)
+
+        #
+        # Check that what we've created is compressed as expected.
+        #
+        with smart_open.open(key, "rb", **no_compression_kwargs) as fin:
+            assert decompressor(fin.read()) == _RAW_DATA
+
+        #
+        # We should be able to read it back as well.
+        #
+        with smart_open.open(key, "rb", **compression_kwargs) as fin:
+            assert fin.read() == _RAW_DATA
+
+    # compression | ignore_ext | behavior |
+    # ----------- | ---------- | -------- |
+    # None        | False      | Enable   |
+    # None        | True       | Disable  |
+
+    @parameterizedtestcase.ParameterizedTestCase.parameterize(
+        ("_compression", "decompressor", "kwargs"),
+        [
+            (
+                "gz",
+                gzip.decompress,
+                dict(ignore_ext=True),
+            ),
+            (
+                "bz2",
+                bz2.decompress,
+                dict(ignore_ext=True),
+            ),
+        ],
+    )
+    @mock_s3
+    def test_rw_compression_by_extension_deprecated(
+        self, _compression, decompressor, kwargs
+    ):
+        """Should read/write files with `_compression`, implicitly inferred by file extension."""
+        s3 = boto3.resource("s3")
+        s3.create_bucket(Bucket="bucket")
+        key = f"s3://bucket/key.{_compression}"
+
+        with smart_open.open(key, "wb") as fout:
+            fout.write(_RAW_DATA)
+
+        #
+        # Check that what we've created is compressed as expected.
+        #
+        with smart_open.open(key, "rb", **kwargs) as fin:
+            assert decompressor(fin.read()) == _RAW_DATA
+
+        #
+        # We should be able to read it back as well.
+        #
+        with smart_open.open(key, "rb") as fin:
```

Comment:
This doesn't need to be part of your tests, because it is not related to compression behavior.

This is round-trip read-write testing. Other tests cover it.

Reply:
- [ ] Remove assertions